operators/src/plot/scatter_plot.rs

```diff
@@ -0,0 +1,1009 @@
+
+/// A scatter plot about two attributes of a vector dataset. If the
+/// dataset contains more then `SCATTER_PLOT_THRESHOLD` elements, this
+/// operator creates a 2D histogram.
+pub type ScatterPlot = Operator<ScatterPlotParams, SingleVectorSource>;
+
+/// The parameter spec for `ScatterPlot`
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct ScatterPlotParams {
+    /// Name of the (numeric) attribute for the x-axis.
+    pub column_x: String,
+    /// Name of the (numeric) attribute for the y-axis.
+    pub column_y: String,
+}
+
+#[typetag::serde]
+#[async_trait]
+impl PlotOperator for ScatterPlot {
+    async fn initialize(
+        self: Box<Self>,
+        context: &dyn ExecutionContext,
+    ) -> Result<Box<dyn InitializedPlotOperator>> {
+        let source = self.sources.vector.initialize(context).await?;
+        for cn in [&self.params.column_x, &self.params.column_y] {
+            match source.result_descriptor().columns.get(cn.as_str()) {
+                Some(column) if !column.is_numeric() => {
+                    return Err(Error::InvalidOperatorSpec {
+                        reason: format!("Column '{}' is not numeric.", cn),
+                    });
+                }
+                Some(_) => {
+                    // OK
+                }
+                None => {
+                    return Err(Error::ColumnDoesNotExist {
+                        column: cn.to_string(),
+                    });
+                }
+            }
+        }
+        Ok(InitializedScatterPlot::new(PlotResultDescriptor {}, self.params, source).boxed())
+    }
+}
+
+/// The initialization of `Histogram`
+pub struct InitializedScatterPlot<Op> {
+    result_descriptor: PlotResultDescriptor,
+    column_x: String,
+    column_y: String,
+    source: Op,
+}
+
+impl<Op> InitializedScatterPlot<Op> {
+    pub fn new(
+        result_descriptor: PlotResultDescriptor,
+        params: ScatterPlotParams,
+        source: Op,
+    ) -> Self {
+        Self {
+            result_descriptor,
+            column_x: params.column_x,
+            column_y: params.column_y,
+            source,
+        }
+    }
+}
+impl InitializedPlotOperator for InitializedScatterPlot<Box<dyn InitializedVectorOperator>> {
+    fn result_descriptor(&self) -> &PlotResultDescriptor {
+        &self.result_descriptor
+    }
+
+    fn query_processor(&self) -> Result<TypedPlotQueryProcessor> {
+        let processor = ScatterPlotQueryProcessor {
+            input: self.source.query_processor()?,
+            column_x: self.column_x.clone(),
+            column_y: self.column_y.clone(),
+        };
+
+        Ok(TypedPlotQueryProcessor::JsonVega(processor.boxed()))
+    }
+}
+
+/// A query processor that calculates the scatter plot about its vector input.
+pub struct ScatterPlotQueryProcessor {
+    input: TypedVectorQueryProcessor,
+    column_x: String,
+    column_y: String,
+}
```

Comment:
Would it be a large overhead to allow two rasters as input?

Michael wrote some temporal alignment adapter and when we query the raster with the same bbox, they are spatially aligned.

Maybe we should also talk to Johannes if this operation is useful.

In any case, maybe it could be also a different task.

Reply:
We made this a new task: https://trello.com/c/viUYH237/466-scatter-plot-f%C3%BCr-2-raster